test/topology_custom/test_mv_backlog.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+import requests
+import re
+
+logger = logging.getLogger(__name__)
+
+async def wait_for_view(cql, name, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' AND view_name = '{name}' ALLOW FILTERING")
+        if done[0][0] == node_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+def get_metrics(addr):
+    # The Prometheus API is on port 9180, and always http
+    addr = 'http://' + addr + ':9180/metrics'
+    resp = requests.get(addr)
+    if resp.status_code != 200:
+        pytest.skip('Metrics port 9180 is not available')
+    response = requests.get(addr)
+    assert response.status_code == 200
+    return response.text
+
+def get_metric(metrics, name, requested_labels=None):
+    the_metrics = get_metrics(metrics)
+    total = 0.0
+    lines = re.compile('^'+name+'{.*$', re.MULTILINE)
+    for match in re.findall(lines, the_metrics):
+        a = match.split()
+        metric = a[0]
+        val = float(a[1])
+        # Check if match also matches the requested labels
+        if requested_labels:
+            # we know metric begins with name{ and ends with } - the labels
+            # are what we have between those
+            got_labels = metric[len(name)+1:-1].split(',')
+            # Check that every one of the requested labels is in got_labels:
+            for k, v in requested_labels.items():
+                if not f'{k}="{v}"' in got_labels:
+                    # No match for requested label, skip this metric (python
+                    # doesn't have "continue 2" so let's just set val to 0...
+                    val = 0
+                    break
+        total += float(val)
+    return total
+
+def get_replicas(cql, key):
+    return cql.cluster.metadata.get_replicas("ks", Int32Type.serialize(key, cql.cluster.protocol_version))
+
+async def enable_on_servers(manager, servers, injection, oneshot=False):
+    errs = [manager.api.enable_injection(s.ip_addr, injection, oneshot) for s in servers]
+    return await asyncio.gather(*errs)
+async def disable_on_servers(manager, servers, injection):
+    errs = [manager.api.disable_injection(s.ip_addr, injection) for s in servers]
+    return await asyncio.gather(*errs)
+
+def get_remote_key(cql, local_node):
+    i = 0
+    while local_node == get_replicas(cql, i)[0]:
+        i = i + 1
+    return i
```

Comment:
All of these look like good candidates to be put into the utility module.

Reply:
I moved the `wait_for_view` to utility, I managed to remove the metrics code completely, as we already have access to them through manager.metrics